backend/app/store/search/internal/multiplexer.go

```diff
@@ -0,0 +1,190 @@
+package internal
+
+import (
+	"context"
+	"sync/atomic"
+	"time"
+
+	log "github.com/go-pkgz/lgr"
+	"github.com/hashicorp/go-multierror"
+	"github.com/pkg/errors"
+
+	"github.com/umputun/remark42/backend/app/store"
+	"github.com/umputun/remark42/backend/app/store/engine"
+	service "github.com/umputun/remark42/backend/app/store/search/service"
+)
+
+// multiplexer handles search requests siteID to particular engine
+type multiplexer struct {
+	shards     map[string]*bufferedEngine
+	engineType string
+	ready      atomic.Value
+}
+
+func newMultiplexer(shards map[string]*bufferedEngine, engineType string) *multiplexer {
+	m := &multiplexer{
+		shards:     shards,
+		engineType: engineType,
+	}
+	m.ready.Store(false)
+	return m
+}
+
+// IndexDocument adds comment to index
+func (s *multiplexer) IndexDocument(comment *store.Comment) error {
+	searcher, has := s.shards[comment.Locator.SiteID]
+	if !has {
+		return errors.Errorf("no search index for site %q", comment.Locator.SiteID)
+	}
+	doc := service.DocFromComment(comment)
+	return searcher.IndexDocument(doc)
+}
+
+const maxErrsDuringStartup = 20
+
+// Init creates missing indexes and index existing documents
+func (s *multiplexer) Init(ctx context.Context, e engine.Interface) error {
+	/* TODO(@vdimir)
```

Comment:
Actually now we call `Init` in synchronous way (https://github.com/umputun/remark42/pull/757#discussion_r467503439), so new document won't be handled during `Init`

Reply:
So this is not actual anymore?

Reply:
To clarify what I mean: this function should be removed from the current code as far as I can see, that's the only action item in that thread.

Reply:
It's required to initialize searcher at startup, read non-indexer documents, and important for first start. How it can be removed?

Reply:
Sorry, I misunderstood you and the code. Nothing to do here then.